pkg/config/app_configuration.go

```diff
@@ -13,6 +13,13 @@ type ApplicationConfig struct {
 	// Duration. example: "30s"
 	ActorScanInterval string `json:"actorScanInterval"`
 	// Duration. example: "30s"
-	DrainOngoingCallTimeout string `json:"drainOngoingCallTimeout"`
-	DrainRebalancedActors   bool   `json:"drainRebalancedActors"`
+	DrainOngoingCallTimeout string           `json:"drainOngoingCallTimeout"`
+	DrainRebalancedActors   bool             `json:"drainRebalancedActors"`
+	Reentrancy              ReentrancyConfig `json:"reentrancy,omitempty"`
+}
+
+type ReentrancyConfig struct {
+	Enabled bool     `json:"enabled"`
+	Methods []string `json:"methods,omitempty"`
+	Limit   *int     `json:"limit,omitempty"`
```

Comment:
Rename this to something less ambiguous. Is it max stack depth?

Reply:
Yes it's that, renamed it to something that's more direct though I'm not sure how much I like it still. Let me know what you think and we can go from there.